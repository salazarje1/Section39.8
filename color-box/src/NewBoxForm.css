form {
    display: flex;
    align-items: center;
    justify-content: space-around;
    width: 50vw;
    margin: 0 auto;
}