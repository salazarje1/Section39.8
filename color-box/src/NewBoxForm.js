import React, { useState } from 'react'; 
import './NewBoxForm.css'; 

const NewBoxForm = ({ addBox }) => {
    const INITIALSTATE = {
        color: '',
        width: '',
        height: ''
    }

    const [formData, setFormData] = useState(INITIALSTATE);

    const handleChange = (e) => {
         const { name, value } = e.target;
         setFormData(formData => ({
            ...formData,
            [name]: value
         }))
    }

    const handleSubmit = (e) => {
        e.preventDefault(); 
        addBox({ ...formData });
        setFormData(INITIALSTATE); 
    }

    return (
        <form onSubmit={handleSubmit}>
            <div>
                <label htmlFor='color'>Color</label>
                <input 
                    type="text"
                    name="color"
                    id="color"
                    placeholder='color'
                    value={formData.color}
                    onChange={handleChange}
                />
            </div>
            <div>
                <label htmlFor='width'>Width</label>
                <input
                    type="number"
                    min="1"
                    max="10"
                    name="width"
                    id="width"
                    placeholder='width'
                    value={formData.width}
                    onChange={handleChange}
                />
            </div>
            <div>
            <label htmlFor='height'>height</label>
            <input
                type="number"
                min="1"
                max="10"
                name="height"
                id="height"
                placeholder='height'
                value={formData.height}
                onChange={handleChange}
            />
            </div>
            <button>Add Box</button>
        </form>
    )
}


export default NewBoxForm; 