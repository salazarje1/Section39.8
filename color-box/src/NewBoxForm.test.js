import { render } from '@testing-library/react'; 
import NewBoxForm from './NewBoxForm';


test('form renders without crashing', () => {
    render(<NewBoxForm />)
})

test('it matches snapshot', () => {
    const { asFragment } = render(<NewBoxForm />);

    expect(asFragment()).toMatchSnapshot(); 
})