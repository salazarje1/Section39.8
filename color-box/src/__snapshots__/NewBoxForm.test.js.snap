// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <div>
      <label
        for="color"
      >
        Color
      </label>
      <input
        id="color"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        max="10"
        min="1"
        name="width"
        placeholder="width"
        type="number"
        value=""
      />
    </div>
    <div>
      <label
        for="height"
      >
        height
      </label>
      <input
        id="height"
        max="10"
        min="1"
        name="height"
        placeholder="height"
        type="number"
        value=""
      />
    </div>
    <button>
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
